network.go

```diff
@@ -38,17 +47,33 @@ type network struct {
 ////////////////
 
 func linkByHwAddr(netHandle *netlink.Handle, hwAddr string) (netlink.Link, error) {
+	if netHandle == nil {
+		return nil, errNoHandle
+	}
+
+	if hwAddr == "" {
+		return nil, errNoMAC
+	}
+
 	links, err := netHandle.LinkList()
 	if err != nil {
 		return nil, err
 	}
 
 	for _, link := range links {
+		if link == nil {
+			continue
+		}
```

Comment:
links is fetched from `netHandle.LinkList()`, is there any situation that link can be nil? 
Or this is just for assurance?

Reply:
Just for assurance. I could remove this check but it benign I think.